docs/parser_plugins.md

```diff
@@ -1 +1,176 @@
+More on the exact "how" later.
+
+Lastly, we will cover tools for point 5 (building the `archive`) and 2 (targeting data fields) in the next subsections.
+Point 5, in particularl, was a developing nuisance, as it required the reconstruction of the already defined target/NOMAD schema.
+That comes across as doing double work: running over the schema once to define it, and a second time to initialize it.
+The main hurdle to automation here was the variable structure of the source data:
+a well-defined hierarchical format does not ensure that individual file structures remains constant.
+Some leaf nodes or branches may simply not be present.
+
+With all of these points covered, developing a parser now comes down to just implementing point 3 (matching source with target).
+This is the stage where data actually becomes semantically enriched and standardized.
+This is a profoundly human task (barring any further leaps in LLM capabilities), where domain expertise is most needed.
+It is thus this scientific curation task that is now most emphasized in parser development, rather than the coding aspects.
+
+Let us cover the remaining tools, so we can move on to the real science.
+
+### MappingParser
+
+We conceptualize format conversion as restructuring a _hierarchical data tree_.
+For those more familiar with graph theory, this is a _directed acyclic graph_.
+
+Since we always consider two formats, we should think of restructuring as a kind of mapping from source to target branches.
+This mapping is, abstractly enough, itself a hierarchical tree, where the leafs denote the source-to-target conversion.
+We construct our mapping tree from branches, called `Mappers`, that starting from the root, can be iteratively nested.
+The leaf nodes, meanwhile, can be of either two types:
+
+- `Map`: the default mapping from as source leaf/branch to a target leaf/branch. Both of these are expressed as Paths according to the [jmespath specifications]().
+Hint: paths may be relative wrt a parent. The format is  to (a) start the path with the connector, i.e. dot (.); and (b) denote the parent's Path separately.
+
+- `Evaluate`: the more active counterpart of `Map`. It inserts a function (its name given as a string) between the mapping. It passes along a list of source paths -denoted just as in `Map`- as the respective positional arguments, i.e. `*args`.
+
+### Parsing Hierarchical Tree Formats
+
+### Workflows and Super-structures
+
+Now that we have covered the SCF and standard outputs, we have to plant these 
+
+## From Text to Hierarchies
+
+Contrary to XML or HDF5, text files do not come with a machine-readable structure.
+Therefore, step 2 (targeting data fields) requires an intermediate step to fall back on _key matching_.
+The source format is converted to a temporary tree format stored just in RAM by matching lines via regular expressions.
```

Comment:
> The source format is converted to a temporary tree format stored just in RAM by matching lines via regular expressions.

The source format is converted to a temporary tree format stored ~~just~~ in ~~RAM~~ memory by matching lines via regular expressions.